core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -872,9 +872,9 @@ class CompactEntityQuery(driverComponent: DriverComponent)
       val replaceRefsSql = sql"""JSON_REPLACE(attributes,
                                               $slickRefsPath,
                                               CAST(
-                                              REPLACE(
+                                              REGEXP_REPLACE(
                                                 REGEXP_REPLACE(JSON_EXTRACT(attributes, $slickRefsPath), $regex, ''),
-                                                ', ]',
+                                                ',[:space:]+\\]',
```

Comment:
```suggestion
                                                ',[:space:]*\\]',
```
just in case the json is ever written more compactly